tests/build/jvm-build.go

```diff
@@ -0,0 +1,352 @@
+		Expect(err).NotTo(HaveOccurred(), "error creating maven task")
+
+		pipelineLocation := "https://raw.githubusercontent.com/redhat-appstudio/jvm-build-service/main/hack/examples/pipeline.yaml"
+		if len(prOwner) > 0 && len(prCommit) > 0 {
+			pipelineLocation = fmt.Sprintf("https://raw.githubusercontent.com/%s/jvm-build-service/%s/hack/examples/pipeline.yaml", prOwner, prCommit)
+		}
+		klog.Infof("fetching pipeline def from %s", pipelineLocation)
+		pipelineData := streamTektonYaml(pipelineLocation)
+		pipeline := &v1beta1.Pipeline{}
+		err = runtime.DecodeInto(decoder, pipelineData, pipeline)
+		Expect(err).NotTo(HaveOccurred(), "error decoding pipeline")
+
+		err = f.TektonController.DeletePipeline(pipeline.Name, appStudioE2EApplicationsNamespace)
+		if err != nil && !errors.IsNotFound(err) {
+			Expect(err).NotTo(HaveOccurred(), "error cleaning up build pipeline before create")
+		}
+		_, err = f.TektonController.CreatePipeline(pipeline, appStudioE2EApplicationsNamespace)
+		Expect(err).NotTo(HaveOccurred(), "error creating build pipeline")
+
+		runLocation := "https://raw.githubusercontent.com/redhat-appstudio/jvm-build-service/main/hack/examples/run.yaml"
+		if len(prOwner) > 0 && len(prCommit) > 0 {
+			runLocation = fmt.Sprintf("https://raw.githubusercontent.com/%s/jvm-build-service/%s/hack/examples/run.yaml", prOwner, prCommit)
+		}
+		klog.Infof("fetching pipelinerun def from %s", runLocation)
+		runData := streamTektonYaml(runLocation)
+		run := &v1beta1.PipelineRun{}
+		err = runtime.DecodeInto(decoder, runData, run)
+		Expect(err).NotTo(HaveOccurred(), "error decoding build pipelinerun")
+
+		// since we use generated name no need to cleanup before create
+		run, err = f.TektonController.CreatePipelineRun(run, appStudioE2EApplicationsNamespace)
+		Expect(err).NotTo(HaveOccurred(), "error creating build pipelinerun")
+		prGeneratedName = run.Name
+
+		klog.Infof("Generated pipeline run %s", prGeneratedName)
+	})
+
+	When("pipelinerun with the appropriate jvm-build-service repository analysis are launched", func() {
+
+		It("those pipeline runs complete successfully", func() {
+			Eventually(func() bool {
+				pr, err := f.TektonController.GetPipelineRun(prGeneratedName, appStudioE2EApplicationsNamespace)
+				if err != nil {
+					klog.Infof("get of pr %s returned error: %s", prGeneratedName, err.Error())
+					return false
+				}
+				if !pr.IsDone() {
+					prBytes, err := json.MarshalIndent(pr, "", "  ")
+					if err != nil {
+						klog.Infof("problem marshalling in progress pipelinerun to bytes: %s", err.Error())
+						return false
+					}
+					klog.Infof("in flight pipeline run: %s", string(prBytes))
+				}
+				prLogs, err := f.TektonController.GetPipelineRunLogs(pr.Name, pr.Namespace)
+				if err != nil {
+					klog.Infof("only partial logs at best for pipeline run %s: err: %s logs: %s", pr.Name, err.Error(), prLogs)
+				} else {
+					klog.Infof("pipeline run logs for %s: %s", pr.Name, prLogs)
```

Comment:
hmm ... yeah seems like a bug.  I'll investigate.

It is for debugging.  I tend to be more verbose with my test logging.  I like to have the logs there even if the test thinks it was "successful" in case I see something amiss in the logs that the test is not catching.

But if my logging verbosity is too counter to how you all want things to work in this repo @psturc I can see about storing such data in a buffer and only print it if the test fails.  A tad more complexity but not too bad.

Reply:
yeah I found at least one problem ... attempting a fix now, will test locally when my cluster for today is ready